packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -529,6 +529,43 @@ export function isTaggedTelemetryPropertyValue(x: any): x is ITaggedTelemetryPro
 
 export const isILoggingError = (x: any): x is ILoggingError => typeof x?.getTelemetryProperties === "function";
 
+/**
+ * Read-Write Logging Error.  Not exported.
+ * This type alias includes addTelemetryProperties, and applies to objects even if not instanceof LoggingError\
+ */
+type RwLoggingError = LoggingError;
+
+/** Type guard for RwLoggingError.  Not exported. */
+const isRwLoggingError = (x: any): x is RwLoggingError =>
+    typeof x?.addTelemetryProperties === "function" && isILoggingError(x);
+
+/**
+ * Annotate the given error object with the given logging props
+ * @returns The same error object passed in if possible, with telemetry props functionality mixed in
+ */
+export function annotateError(
+    error: unknown,
+    props: ITelemetryProperties,
+): ILoggingError {
+    if (isRwLoggingError(error)) {
+        error.addTelemetryProperties(props);
+        return error;
+    }
+
+    if (isRegularObject(error)) {
+        // Even though it's not exposed, fully implement IRwLoggingError for subsequent calls to annotateError
+        const loggingError = error as RwLoggingError;
+
+        const propsForError = {...props};
+        loggingError.getTelemetryProperties = () => propsForError;
+        loggingError.addTelemetryProperties = (p: ITelemetryProperties) => { Object.assign(propsForError, p); };
+        return loggingError;
+    }
+
+    const message = String(error);
+    return new LoggingError(message, props);
```

Comment:
can we remove the create case from this method, or have one that doesn't have it? errors should always be objects anyway

Reply:
For context, here's a preview of the next change I'll make, to enhance `annotateError` to yield a proper `IFluidErrorBase`:

https://github.com/markfields/FluidFramework/pull/1